opentracing-api/src/main/java/io/opentracing/Tracer.java

```diff
@@ -20,113 +20,154 @@
-
-  /**
-   * Extract a SpanContext from a `carrier` of a given type, presumably after propagation across a process boundary.
-   *
-   * <p>Example:
-   * <pre>{@code
-   * Tracer tracer = ...
-   * TextMap httpHeadersCarrier = new AnHttpHeaderCarrier(httpRequest);
-   * SpanContext spanCtx = tracer.extract(Format.Builtin.HTTP_HEADERS, httpHeadersCarrier);
-   * tracer.buildSpan('...').asChildOf(spanCtx).start();
-   * }</pre>
-   *
-   * If the span serialized state is invalid (corrupt, wrong version, etc) inside the carrier this will result in an
-   * IllegalArgumentException.
-   *
-   * @param <C> the carrier type, which also parametrizes the Format.
-   * @param format the Format of the carrier
-   * @param carrier the carrier for the SpanContext state. All Tracer.extract() implementations must support
-   *                io.opentracing.propagation.TextMap and java.nio.ByteBuffer.
-   *
-   * @return the SpanContext instance holding context to create a Span.
-   *
-   * @see io.opentracing.propagation.Format
-   * @see io.opentracing.propagation.Format.Builtin
-   */
-  <C> SpanContext extract(Format<C> format, C carrier);
-
-
-  interface SpanBuilder extends SpanContext {
-
-      /**
-       * A shorthand for addReference(References.CHILD_OF, parent).
-       */
-      SpanBuilder asChildOf(SpanContext parent);
-
-      /**
-       * A shorthand for addReference(References.CHILD_OF, parent.context()).
-       */
-      SpanBuilder asChildOf(Span parent);
-
-      /**
-       * Add a reference from the Span being built to a distinct (usually parent) Span. May be called multiple times to
-       * represent multiple such References.
-       *
-       * @param referenceType the reference type, typically one of the constants defined in References
-       * @param referencedContext the SpanContext being referenced; e.g., for a References.CHILD_OF referenceType, the
-       *                          referencedContext is the parent
-       *
-       * @see io.opentracing.References
-       */
-      SpanBuilder addReference(String referenceType, SpanContext referencedContext);
-
-      /** Same as {@link Span#setTag(String, String)}, but for the span being built. */
-      SpanBuilder withTag(String key, String value);
-
-      /** Same as {@link Span#setTag(String, boolean)}, but for the span being built. */
-      SpanBuilder withTag(String key, boolean value);
-
-      /** Same as {@link Span#setTag(String, Number)}, but for the span being built. */
-      SpanBuilder withTag(String key, Number value);
-
-      /** Specify a timestamp of when the Span was started, represented in microseconds since epoch. */
-      SpanBuilder withStartTimestamp(long microseconds);
-
-      /** Returns the started Span. */
-      Span start();
-
-  }
+    /**
+     * Return a new SpanBuilder for a Span with the given `operationName`.
+     *
+     * <p>If there is an active Span according to the {@link Tracer#scheduler()}'s
```

Comment:
**This one comment is the point of this entire PR.** It would make the life of application (and library) instrumentors so much better+easier+etc. IMO a very exciting improvement for OT-Java and OT in general.